### PR TITLE
Replacing agent namespace with workflow manifest namespace

### PR DESCRIPTION
**Pull request by rajdas98**
Signed-off-by: Raj Babu Das <raj.das@mayadata.io>

## Proposed changes

Replacing agent namespace with workflow manifest namespace

## Types of changes

What types of changes does your code introduce to Litmus? Put an `x` in the boxes that apply
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices applies)

## Checklist

Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code.
- [ ] I have read the [CONTRIBUTING](https://github.com/litmuschaos/litmus/blob/master/CONTRIBUTING.md) doc
- [ ] I have signed the commit for DCO to be passed.
- [ ] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [ ] I have added necessary documentation (if appropriate)

## Dependency
- Please add the links to the dependent PR need to be merged before this (if any).

### CODE REVIEW

File: litmus-portal/graphql-server/pkg/graphql/mutations/mutation.go [hunk 1]
```diff
@@ -307,11 +317,18 @@ func UpdateWorkflow(workflow *model.ChaosWorkFlowInput, r store.StateData) (*mod
 		return nil, err
 	}
 
+	workflowNamespace := gjson.Get(workflow.WorkflowManifest, "metadata.namespace").String()
```
Comment (gdsoumya) [inline litmus-portal/graphql-server/pkg/graphql/mutations/mutation.go:320]: Is there a need to add a new package? json.Unmarshal would work too?

Comment (rajdas98) [reply litmus-portal/graphql-server/pkg/graphql/mutations/mutation.go:320]: WorkflowManifest is JSON string and to unMarshal WorkflowManifest we need a struct of workflow manifest

Comment (gdsoumya) [reply litmus-portal/graphql-server/pkg/graphql/mutations/mutation.go:320]: We can use map[string]inteface{} to locate the "namespace" key, if it is there we use it or else we use the os.Getenv()

Comment (rajdas98) [reply litmus-portal/graphql-server/pkg/graphql/mutations/mutation.go:320]: Yes, we can use but just to ease the getting of the namespace. I have used it. 
We have used similar pkg for setting also

Comment (rajdas98) [reply litmus-portal/graphql-server/pkg/graphql/mutations/mutation.go:320]: FYI map[string]map[string]interface{} or map[string]interface{} is not able to parse the manifest

Comment (gdsoumya) [reply litmus-portal/graphql-server/pkg/graphql/mutations/mutation.go:320]: Why is map[string]interface{} not able to parse it? I remember parsing it while I made the changes for the wf_id in cron-workflows?

Comment (gdsoumya) [reply litmus-portal/graphql-server/pkg/graphql/mutations/mutation.go:320]: We can use this package, i think Ishan already used the sjson package to add the wf_id.

Comment (rajdas98) [reply litmus-portal/graphql-server/pkg/graphql/mutations/mutation.go:320]: You have use sjson to set wf_id in cron-workflow, For parsing this kind of JSON, it requires proper struct.

Comment (gdsoumya) [reply litmus-portal/graphql-server/pkg/graphql/mutations/mutation.go:320]: No I did not use sjson, it was already there I would've used simple map for that also. 

This is the unmarshaling code I used 
```
var workflow map[string]interface{}
	err := json.Unmarshal([]byte(input.WorkflowManifest), &workflow)
	if err != nil {
		return nil, err
	}
```

Comment (gdsoumya) [reply litmus-portal/graphql-server/pkg/graphql/mutations/mutation.go:320]: To get the ns you would have to marshal twice maybe, so it's easier to use this gjson package

Comment (rajdas98) [reply litmus-portal/graphql-server/pkg/graphql/mutations/mutation.go:320]: Okay, Is it good to merge then?

**Review by gdsoumya — APPROVED** [review]

**Pull request merged**